Better error messages thanks to compile time type names

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26482 [wip] Error checking for KernelFunction::callUnboxed()
* **#30712 Better error messages thanks to compile time type names**
* #31307 [no-land] try fix 2
* #31306 [no-land] try fix 1
* #26628 Remove CAFFE_KNOWN_TYPE

Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

smessmer committed:
[wip] Better error messages thanks to compile time type names



Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

ezyang commented on aten/src/ATen/core/jit_type.h:
```diff
@@ -1246,7 +1246,10 @@ template <typename T>
   static TypePtr call() {
     if (!isCustomClassRegistered<T>()) {
-      throw c10::Error("Type could not be converted to any of the known types.", "");
+      throw c10::Error(
```
Why aren't you using `TORCH_CHECK(0, msg...)` here?

kostmo commented:
## CircleCI build failures summary
As of commit 4123a78:
* **3/12** broken upstream at merge base baccd26 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=baccd26df7e7c0d808b740dae93d404c48b6d389&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * Since your merge base is older than `viable/strict`, run these commands:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **9/12** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=4123a78859f6e58a27aba0d15894b7dde31d8f12).

### 9 new failures recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3909892) binary_linux_manywheel_2_7mu_cpu_devtoolset7_build (1/9)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=72561301))
```
Dec 11 22:42:14 /pytorch/c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = float]’
Dec 11 22:42:14 /pytorch/c10/core/DefaultDtype.cpp:5:59:   required from here
Dec 11 22:42:14 /pytorch/c10/util/typeid.h:72:22: note:   initializing argument 1 of ‘constexpr caffe2::TypeIdentifier::TypeIdentifier(c10::util::type_index)’
Dec 11 22:42:14    constexpr explicit TypeIdentifier(c10::util::type_index id) : IdWrapper(id) {}
Dec 11 22:42:14                       ^~~~~~~~~~~~~~
Dec 11 22:42:14 /pytorch/c10/util/typeid.h: In instantiation of ‘caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = float]’:
Dec 11 22:42:14 /pytorch/c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = float]’
Dec 11 22:42:14 /pytorch/c10/core/DefaultDtype.cpp:5:59:   required from here
Dec 11 22:42:14 /pytorch/c10/util/typeid.h:282:62:   in constexpr expansion of ‘caffe2::TypeIdentifier::Get<float>()’
Dec 11 22:42:14 /pytorch/c10/util/typeid.h:64:55:   in constexpr expansion of ‘c10::util::get_type_index<float>()’
Dec 11 22:42:14 /pytorch/c10/util/typeid.h:282:31: error: constexpr call flows off the end of the function
Dec 11 22:42:14    C10_HOST_CONSTEXPR_VAR auto typeId = TypeIdentifier::Get<T>();
Dec 11 22:42:14                                ^~~~~~
Dec 11 22:42:14 [  9%] Generating python/ideep/concat_split_op_test.py
Dec 11 22:42:14 gmake[2]: *** [c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o] Error 1
Dec 11 22:42:14 gmake[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2
Dec 11 22:42:14 gmake[1]: *** Waiting for unfinished jobs....
Dec 11 22:42:14 [  9%] Generating python/ideep/conv_op_test.py
Dec 11 22:42:14 [  9%] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/Fbgemm.cc.o
Dec 11 22:42:14 [  9%] Generating python/ideep/conv_transpose_test.py
Dec 11 22:42:14 [  9%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf-lite.dir/__/src/google/protobuf/message_lite.cc.o
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3909886) binary_linux_libtorch_2_7m_cpu_devtoolset7_shared-with-deps_build (2/9)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=72562329))
```
Dec 11 22:43:11 /pytorch/c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = float]’
Dec 11 22:43:11 /pytorch/c10/core/DefaultDtype.cpp:5:59:   required from here
Dec 11 22:43:11 /pytorch/c10/util/typeid.h:72:22: note:   initializing argument 1 of ‘constexpr caffe2::TypeIdentifier::TypeIdentifier(c10::util::type_index)’
Dec 11 22:43:11    constexpr explicit TypeIdentifier(c10::util::type_index id) : IdWrapper(id) {}
Dec 11 22:43:11                       ^~~~~~~~~~~~~~
Dec 11 22:43:11 /pytorch/c10/util/typeid.h: In instantiation of ‘caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = float]’:
Dec 11 22:43:11 /pytorch/c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = float]’
Dec 11 22:43:11 /pytorch/c10/core/DefaultDtype.cpp:5:59:   required from here
Dec 11 22:43:11 /pytorch/c10/util/typeid.h:282:62:   in constexpr expansion of ‘caffe2::TypeIdentifier::Get<float>()’
Dec 11 22:43:11 /pytorch/c10/util/typeid.h:64:55:   in constexpr expansion of ‘c10::util::get_type_index<float>()’
Dec 11 22:43:11 /pytorch/c10/util/typeid.h:282:31: error: constexpr call flows off the end of the function
Dec 11 22:43:11    C10_HOST_CONSTEXPR_VAR auto typeId = TypeIdentifier::Get<T>();
Dec 11 22:43:11                                ^~~~~~
Dec 11 22:43:11 gmake[2]: *** [c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o] Error 1
Dec 11 22:43:11 gmake[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2
Dec 11 22:43:11 gmake[1]: *** Waiting for unfinished jobs....
Dec 11 22:43:11 [  6%] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC16.cc.o
Dec 11 22:43:12 [  7%] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/GenerateKernelU8S8S32ACC16Avx512.cc.o
Dec 11 22:43:12 [  7%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/empty.pb.cc.o
Dec 11 22:43:12 [  7%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/extension_set_heavy.cc.o
Dec 11 22:43:12 [  7%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/field_mask.pb.cc.o
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3909876) pytorch_linux_xenial_py2_7_9_build (3/9)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=72563830))
```
Dec 11 22:46:33 /var/lib/jenkins/workspace/c10/util/typeid.h:424:45:   required from 'static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = c10::BFloat16]'
Dec 11 22:46:33 /var/lib/jenkins/workspace/c10/core/ScalarType.h:179:5:   required from here
Dec 11 22:46:33 /var/lib/jenkins/workspace/c10/util/typeid.h:72:22: note:   initializing argument 1 of 'constexpr caffe2::TypeIdentifier::TypeIdentifier(c10::util::type_index)'
Dec 11 22:46:33    constexpr explicit TypeIdentifier(c10::util::type_index id) : IdWrapper(id) {}
Dec 11 22:46:33                       ^~~~~~~~~~~~~~
Dec 11 22:46:33 /var/lib/jenkins/workspace/c10/util/typeid.h: In instantiation of 'caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = c10::BFloat16]':
Dec 11 22:46:33 /var/lib/jenkins/workspace/c10/util/typeid.h:424:45:   required from 'static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = c10::BFloat16]'
Dec 11 22:46:33 /var/lib/jenkins/workspace/c10/core/ScalarType.h:179:5:   required from here
Dec 11 22:46:33 /var/lib/jenkins/workspace/c10/util/typeid.h:282:62:   in constexpr expansion of 'caffe2::TypeIdentifier::Get<c10::BFloat16>()'
Dec 11 22:46:33 /var/lib/jenkins/workspace/c10/util/typeid.h:64:55:   in constexpr expansion of 'c10::util::get_type_index<c10::BFloat16>()'
Dec 11 22:46:33 /var/lib/jenkins/workspace/c10/util/typeid.h:282:31: error: constexpr call flows off the end of the function
Dec 11 22:46:33    C10_HOST_CONSTEXPR_VAR auto typeId = TypeIdentifier::Get<T>();
Dec 11 22:46:33                                ^~~~~~
Dec 11 22:46:33 c10/CMakeFiles/c10.dir/build.make:86: recipe for target 'c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o' failed
Dec 11 22:46:33 make[2]: *** [c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o] Error 1
Dec 11 22:46:33 CMakeFiles/Makefile2:2009: recipe for target 'c10/CMakeFiles/c10.dir/all' failed
Dec 11 22:46:33 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2
Dec 11 22:46:33 make[1]: *** Waiting for unfinished jobs....
Dec 11 22:46:33 [  7%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/transport/address.cc.o
Dec 11 22:46:33 [  7%] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_core_bf16_sum.cpp.o
Dec 11 22:46:33 [  7%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/transport/buffer.cc.o
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3909882) binary_linux_libtorch_2_7m_cpu_gcc5_4_cxx11-abi_shared-with-deps_build (4/9)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=72563847))
```
Dec 11 22:46:08 ../c10/util/TypeIndex.h:24:8: note: candidate: constexpr c10::util::type_index::type_index(const c10::util::type_index&)
Dec 11 22:46:08  struct type_index final : IdWrapper<type_index, uint64_t> {
Dec 11 22:46:08         ^
Dec 11 22:46:08 ../c10/util/TypeIndex.h:24:8: note:   conversion of argument 1 would be ill-formed:
Dec 11 22:46:08 ../c10/util/TypeIndex.h:24:8: note: candidate: constexpr c10::util::type_index::type_index(c10::util::type_index&&)
Dec 11 22:46:08 ../c10/util/TypeIndex.h:24:8: note:   conversion of argument 1 would be ill-formed:
Dec 11 22:46:10 [1544/2980] Building CXX object caffe2/CMakeFiles/inline_container_test.dir/serialize/inline_container_test.cc.o
Dec 11 22:46:10 [1545/2980] Building CXX object caffe2/CMakeFiles/TensorImpl_test.dir/__/aten/src/ATen/core/TensorImpl_test.cpp.o
Dec 11 22:46:10 [1546/2980] Building CXX object caffe2/CMakeFiles/KernelFunction_test.dir/__/aten/src/ATen/core/boxing/KernelFunction_test.cpp.o
Dec 11 22:46:11 [1547/2980] Building CXX object caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o
Dec 11 22:46:11 FAILED: caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o 
Dec 11 22:46:11 /usr/bin/c++  -DAT_PARALLEL_OPENMP=1 -DHAVE_MALLOC_USABLE_SIZE=1 -DHAVE_MMAP=1 -DHAVE_SHM_OPEN=1 -DHAVE_SHM_UNLINK=1 -DIDEEP_USE_MKL -DONNX_ML=1 -DONNX_NAMESPACE=onnx_torch -DTH_BLAS_MKL -D_FILE_OFFSET_BITS=64 -Dcaffe2_pybind11_state_EXPORTS -Iaten/src -I../aten/src -I. -I../ -I../cmake/../third_party/benchmark/include -Icaffe2/contrib/aten -I../third_party/onnx -Ithird_party/onnx -I../third_party/foxi -Ithird_party/foxi -Icaffe2/aten/src/TH -I../aten/src/TH -Icaffe2/aten/src -I../aten/../third_party/catch/single_include -I../aten/src/ATen/.. -Icaffe2/aten/src/ATen -I../third_party/miniz-2.0.8 -I../caffe2/core/nomnigraph/include -I../caffe2/../torch/csrc/api -I../caffe2/../torch/csrc/api/include -I../c10/.. -Ithird_party/ideep/mkl-dnn/include -I../third_party/ideep/mkl-dnn/src/../include -isystem third_party/gloo -isystem ../cmake/../third_party/gloo -isystem ../cmake/../third_party/googletest/googlemock/include -isystem ../cmake/../third_party/googletest/googletest/include -isystem ../third_party/protobuf/src -isystem /opt/intel/include -isystem ../third_party/gemmlowp -isystem ../third_party/neon2sse -isystem ../third_party -isystem ../cmake/../third_party/eigen -isystem /opt/conda/include/python2.7 -isystem /opt/conda/lib/python2.7/site-packages/numpy/core/include -isystem ../cmake/../third_party/pybind11/include -isystem /opt/rocm/hip/include -isystem /include -isystem ../third_party/ideep/mkl-dnn/include -isystem ../third_party/ideep/include -D_GLIBCXX_USE_CXX11_ABI=1 -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -DHAVE_AVX_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION -O3  -fPIC   -fvisibility=hidden -DCAFFE2_USE_GLOO -DHAVE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -fopenmp -std=gnu++14 -MD -MT caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o -MF caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o.d -o caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_ideep.cc.o -c ../caffe2/python/pybind_state_ideep.cc
Dec 11 22:46:11 In file included from ../c10/util/typeid.h:27:0,
Dec 11 22:46:11                  from ../caffe2/core/context_base.h:11,
Dec 11 22:46:11                  from ../caffe2/core/context.h:10,
Dec 11 22:46:11                  from ../caffe2/python/pybind_state.h:5,
Dec 11 22:46:11                  from ../caffe2/python/pybind_state_ideep.cc:7:
Dec 11 22:46:11 ../c10/util/TypeIndex.h: In instantiation of ‘constexpr c10::util::type_index c10::util::get_type_index() [with T = std::__cxx11::basic_string<char>]’:
Dec 11 22:46:11 ../c10/util/typeid.h:64:55:   required from ‘static constexpr caffe2::TypeIdentifier caffe2::TypeIdentifier::Get() [with T = std::__cxx11::basic_string<char>]’
Dec 11 22:46:11 ../c10/util/typeid.h:282:62:   required from ‘caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = std::__cxx11::basic_string<char>]’
Dec 11 22:46:11 ../c10/util/typeid.h:424:45:   required from ‘static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = std::__cxx11::basic_string<char>]’
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3909875) caffe2_onnx_py2_gcc5_ubuntu16_04_build (5/9)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=72581857))
```
Dec 11 23:03:36                  from /var/lib/jenkins/workspace/caffe2/core/blob.h:11,
Dec 11 23:03:36                  from /var/lib/jenkins/workspace/caffe2/core/blob_serialization.h:9,
Dec 11 23:03:36                  from /var/lib/jenkins/workspace/caffe2/core/blob_serialization.cc:1:
Dec 11 23:03:36 /var/lib/jenkins/workspace/c10/util/typeid.h: In instantiation of 'caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = short unsigned int]':
Dec 11 23:03:36 /var/lib/jenkins/workspace/c10/util/typeid.h:424:45:   required from 'static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = short unsigned int]'
Dec 11 23:03:36 /var/lib/jenkins/workspace/c10/core/TensorImpl.h:1333:70:   required from 'T* c10::TensorImpl::mutable_data() [with T = short unsigned int]'
Dec 11 23:03:36 /var/lib/jenkins/workspace/caffe2/core/tensor.h:444:41:   required from 'T* caffe2::Tensor::mutable_data() const [with T = short unsigned int]'
Dec 11 23:03:36 /var/lib/jenkins/workspace/caffe2/core/blob_serialization.cc:570:51:   required from here
Dec 11 23:03:36 /var/lib/jenkins/workspace/c10/util/typeid.h:282:62:   in constexpr expansion of 'caffe2::TypeIdentifier::Get<short unsigned int>()'
Dec 11 23:03:36 /var/lib/jenkins/workspace/c10/util/typeid.h:64:55:   in constexpr expansion of 'c10::util::get_type_index<short unsigned int>()'
Dec 11 23:03:36 /var/lib/jenkins/workspace/c10/util/typeid.h:282:63: error: constexpr call flows off the end of the function
Dec 11 23:03:36    C10_HOST_CONSTEXPR_VAR auto typeId = TypeIdentifier::Get<T>();
Dec 11 23:03:36                                                                ^
Dec 11 23:03:36 caffe2/CMakeFiles/torch_cpu.dir/build.make:8219: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/core/blob_serialization.cc.o' failed
Dec 11 23:03:36 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/core/blob_serialization.cc.o] Error 1
Dec 11 23:03:36 make[2]: *** Waiting for unfinished jobs....
Dec 11 23:03:49 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2
Dec 11 23:03:49 CMakeFiles/Makefile2:5422: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed
Dec 11 23:03:49 Makefile:138: recipe for target 'all' failed
Dec 11 23:03:49 make: *** [all] Error 2
Dec 11 23:03:49 Building wheel torch-1.4.0a0+4123a78
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3909874) pytorch_linux_xenial_py3_6_gcc5_4_build (6/9)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=72584952))
```
Dec 11 23:05:41                  from /var/lib/jenkins/workspace/caffe2/core/blob.h:11,
Dec 11 23:05:41                  from /var/lib/jenkins/workspace/caffe2/core/blob_serialization.h:9,
Dec 11 23:05:41                  from /var/lib/jenkins/workspace/caffe2/core/blob_serialization.cc:1:
Dec 11 23:05:41 /var/lib/jenkins/workspace/c10/util/typeid.h: In instantiation of 'caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance() [with T = short unsigned int]':
Dec 11 23:05:41 /var/lib/jenkins/workspace/c10/util/typeid.h:424:45:   required from 'static caffe2::TypeMeta caffe2::TypeMeta::Make() [with T = short unsigned int]'
Dec 11 23:05:41 /var/lib/jenkins/workspace/c10/core/TensorImpl.h:1314:70:   required from 'T* c10::TensorImpl::mutable_data() [with T = short unsigned int]'
Dec 11 23:05:41 /var/lib/jenkins/workspace/caffe2/core/tensor.h:444:41:   required from 'T* caffe2::Tensor::mutable_data() const [with T = short unsigned int]'
Dec 11 23:05:41 /var/lib/jenkins/workspace/caffe2/core/blob_serialization.cc:575:51:   required from here
Dec 11 23:05:41 /var/lib/jenkins/workspace/c10/util/typeid.h:282:62:   in constexpr expansion of 'caffe2::TypeIdentifier::Get<short unsigned int>()'
Dec 11 23:05:41 /var/lib/jenkins/workspace/c10/util/typeid.h:64:55:   in constexpr expansion of 'c10::util::get_type_index<short unsigned int>()'
Dec 11 23:05:41 /var/lib/jenkins/workspace/c10/util/typeid.h:282:63: error: constexpr call flows off the end of the function
Dec 11 23:05:41    C10_HOST_CONSTEXPR_VAR auto typeId = TypeIdentifier::Get<T>();
Dec 11 23:05:41                                                                ^
Dec 11 23:05:41 caffe2/CMakeFiles/torch_cpu.dir/build.make:8267: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/core/blob_serialization.cc.o' failed
Dec 11 23:05:41 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/core/blob_serialization.cc.o] Error 1
Dec 11 23:05:41 make[2]: *** Waiting for unfinished jobs....
Dec 11 23:05:42 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2
Dec 11 23:05:42 CMakeFiles/Makefile2:5273: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed
Dec 11 23:05:42 make: *** [all] Error 2
Dec 11 23:05:42 Makefile:138: recipe for target 'all' failed
Dec 11 23:05:42 -- Building version 1.4.0a0+857f8e7
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3909873) pytorch_windows_build (7/9)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=72618473))
```
C:/Users/circleci/project\c10/util/typeid.h(283): here
            instantiation of "caffe2::detail::TypeMetaData caffe2::detail::_makeTypeMetaDataInstance<T>() [with T=c10::BFloat16]" 
C:/Users/circleci/project\c10/util/typeid.h(424): here
            instantiation of "caffe2::TypeMeta caffe2::TypeMeta::Make<T>() [with T=c10::BFloat16]" 
C:/Users/circleci/project\c10/core/ScalarType.h(179): here

16 errors detected in the compilation of "C:/Users/circleci/AppData/Local/Temp/tmpxft_0000220c_00000000-10_BatchLinearAlgebra.cpp1.ii".
BatchLinearAlgebra.cu
-- Removing C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/./torch_cuda_generated_BatchLinearAlgebra.cu.obj
"C:/Program Files/CMake/bin/cmake.exe" -E remove C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/./torch_cuda_generated_BatchLinearAlgebra.cu.obj
CMake Error at torch_cuda_generated_BatchLinearAlgebra.cu.obj.Release.cmake:281 (message):
  Error generating file
  C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/./torch_cuda_generated_BatchLinearAlgebra.cu.obj


ninja: build stopped: subcommand failed.
-- Building version 1.4.0a0+4123a78
cmake -GNinja -DBUILD_ENVIRONMENT=pytorch-win-ws2019-cuda10-cudnn7-py3 -DBUILD_PYTHON=True -DBUILD_TEST=True -DBUILD_TYPE=release -DCMAKE_BUILD_TYPE=Release -DCMAKE_GENERATOR=Ninja -DCMAKE_INCLUDE_PATH=C:\Users\circleci\project\build\win_tmp\mkl\include -DCMAKE_INSTALL_PREFIX=C:\Users\circleci\project\torch -DCMAKE_PREFIX_PATH=C:\Jenkins\Miniconda3\Lib\site-packages -DCMAKE_VERBOSE_MAKEFILE=1 -DCUDA_NVCC_EXECUTABLE=C:\Users\circleci\project\build\win_tmp\bin\nvcc -DCUDNN_LIBRARY=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\lib\x64 -DJAVA_HOME=C:\Program Files\OpenJDK\jdk-12.0.2 -DNUMPY_INCLUDE_DIR=C:\Jenkins\Miniconda3\lib\site-packages\numpy\core\include -DPYTHON_EXECUTABLE=C:\Jenkins\Miniconda3\python.exe -DPYTHON_INCLUDE_DIR=C:\Jenkins\Miniconda3\include -DPYTHON_LIBRARY=C:\Jenkins\Miniconda3/libs/python36.lib -DTORCH_BUILD_VERSION=1.4.0a0+4123a78 -DUSE_CUDA=1 -DUSE_NUMPY=True C:\Users\circleci\project
cmake --build . --target install --config Release -- -j 15
Traceback (most recent call last):
  File "setup.py", line 755, in <module>
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3912784) pytorch_macos_10_13_py3_test (8/9)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=72695371))
```
Dec 11 17:12:41   File "test_fake_quant.py", line 7, in <module>
Dec 11 17:12:41     import hypothesis_utils as hu
Dec 11 17:12:41   File "/Users/distiller/project/test/hypothesis_utils.py", line 9, in <module>
Dec 11 17:12:41     from hypothesis.searchstrategy import SearchStrategy
Dec 11 17:12:41 ModuleNotFoundError: No module named 'hypothesis.searchstrategy'
Dec 11 17:12:41 Traceback (most recent call last):
Dec 11 17:12:41   File "test/run_test.py", line 456, in <module>
Dec 11 17:12:41     main()
Dec 11 17:12:41   File "test/run_test.py", line 449, in main
Dec 11 17:12:41     raise RuntimeError(message)
Dec 11 17:12:41 RuntimeError: test_fake_quant failed!
Dec 11 17:12:41 + cleanup
Dec 11 17:12:41 + retcode=1
Dec 11 17:12:41 + set +x
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3912844) pytorch_linux_xenial_py3_clang5_asan_test (9/9)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=72697223))
```
Dec 12 01:14:26   File "test_nn.py", line 56, in <module>
Dec 12 01:14:26     import hypothesis_utils as hu
Dec 12 01:14:26   File "/var/lib/jenkins/workspace/test/hypothesis_utils.py", line 9, in <module>
Dec 12 01:14:26     from hypothesis.searchstrategy import SearchStrategy
Dec 12 01:14:26 ModuleNotFoundError: No module named 'hypothesis.searchstrategy'
Dec 12 01:14:26 Traceback (most recent call last):
Dec 12 01:14:26   File "test/run_test.py", line 456, in <module>
Dec 12 01:14:26     main()
Dec 12 01:14:26   File "test/run_test.py", line 449, in main
Dec 12 01:14:26     raise RuntimeError(message)
Dec 12 01:14:26 RuntimeError: test_nn failed!
Dec 12 01:14:26 + cleanup
Dec 12 01:14:26 + retcode=1
Dec 12 01:14:26 + set +x
Dec 12 01:14:26 =================== sccache compilation log ===================
Dec 12 01:14:26 =========== If your build fails, please take a look at the log above for possible reasons ===========
Dec 12 01:14:26 Compile requests                  7
Dec 12 01:14:26 Compile requests executed         6
Dec 12 01:14:26 Cache hits                        0
Dec 12 01:14:26 Cache misses                      6
Dec 12 01:14:26 Cache timeouts                    0
```
### 3 upstream failures recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_linux_xenial_cuda9_cudnn7_py3_build](https://dr.pytorch.org/build-details.html?build_id=72589844)
* [pytorch_libtorch_linux_xenial_cuda9_cudnn7_py3_build](https://dr.pytorch.org/build-details.html?build_id=72590861)
* [pytorch_xla_linux_xenial_py3_6_clang7_build](https://dr.pytorch.org/build-details.html?build_id=72656027)



---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]

smessmer committed:
Update on "Better error messages thanks to compile time type names"


Differential Revision: [D18805082](https://our.internmc.facebook.com/intern/diff/D18805082/)

[ghstack-poisoned]
